### ISSUE TITLE
Reporter - Exception: no such function: floor 

### ISSUE THREAD

**Issue by ArekKuczynski** [damies13/rfswarm#306]
**Describe the bug**
When loading specific Results and Template in Reporter with python 3.9, the db report file cannot be loaded due to this exception.
This bug was found during regression testing Reporter, actions: 
https://github.com/NiceProjectPoland/rfswarm/actions/runs/11129091861
The template section that is involved in this issue:
```
[FB6A07238B2L]
datatype = Result
changed = 1727785420.214303
sql = SELECT floor(end_time) as 'Time' , count(result) as 'Value' , result_name as x91Namex93 FROM Results WHERE x91Namex93 NOT LIKE 'Exception in threadx37' AND end_time >= 1690522973 AND end_time <= 1690523400 GROUP by floor(end_time) , result_name , result ORDER by floor(end_time), result DESC, count(result) DESC
metrictype = 
filteragent = None
filtertype = None
primarymetric = 
secondarymetric = 
resulttype = TPS
filterresult = None
axisen = 1
```
The exception details:
```
ReporterBase: rt_graph_LR_Ids(966): [6:5]	 id: FB6A07238B2 	[1.4.0 @2024-10-01 08:50:16]
ReporterBase: rt_graph_LR_Ids(966): [6:5]	 id: FB6A07238B2 	[1.4.0 @2024-10-01 08:50:17]
ReporterGUI: cp_graph(9248): [6:5]	 pid: FB6A07238B2 	idl: FB6A07238B2L 	idr: FB6A07238B2R 	[1.4.0 @2024-10-01 08:50:17]
ReporterBase: rt_table_get_enfr(1891): [6:5]	 id: FB6A07238B2L 	[1.4.0 @2024-10-01 08:50:17]
ReporterBase: rt_table_get_enfa(1985): [6:5]	 id: FB6A07238B2L 	[1.4.0 @2024-10-01 08:50:17]
ReporterBase: rt_table_get_fa(1915): [6:5]	 id: FB6A07238B2L 	[1.4.0 @2024-10-01 08:50:17]
ReporterBase: rt_graph_generate_sql(1354): [6:5]	 sql: SELECT floor(end_time) as 'Time' , count(result) as 'Value' , result_name as [Name] FROM Results WHERE [Name] NOT LIKE 'Exception in thread%' AND end_time >= 1690522973 AND end_time <= 1690523400 GROUP by floor(end_time) , result_name , result ORDER by floor(end_time), result DESC, count(result) DESC  	[1.4.0 @2024-10-01 08:50:17]
ReporterBase: rt_graph_generate_sql(1354): [6:5]	 sql:  	[1.4.0 @2024-10-01 08:50:17]
ReporterBase: run_db_thread(2604): [6:1]	 run_db_thread: dbqueue: Read: Exception: no such function: floor 	[1.4.0 @2024-10-01 08:50:17]
ReporterBase: run_db_thread(2605): [6:1]	 run_db_thread: dbqueue: Read: Item: {'SQL': "SELECT floor(end_time) as 'Time' , count(result) as 'Value' , result_name as [Name] FROM Results WHERE [Name] NOT LIKE 'Exception in thread%' AND end_time >= 1690522973 AND end_time <= 1690523400 GROUP by floor(end_time) , result_name , result ORDER by floor(end_time), result DESC, count(result) DESC ", 'KEY': 'FB6A07238B2_1727705359.431957'} 	[1.4.0 @2024-10-01 08:50:17]
ReporterGUI: mnu_export_word(10112): [6:5]	 _event: None 	[1.4.0 @2024-10-01 08:59:10]
```
You can see that there is 9 minutes gap between `run_db_thread(2605)` and `mnu_export_word(10112)`

**To Reproduce**
1. Open Reporter with python 3.9
2. Load Results and Template (used template and results are in Tests/Regression/Reporter/testdata/Issue-#36_37_38
3. See error

**Expected behavior**
The reporter should manage this error and produce a report file, just as the newer Python versions do.

**Screenshots**

**Desktop (please complete the following information):**
- OS: windows-latest
- Reporter Version: v1.4.0 newest 30/9/2024

**Additional context**

**Comment by damies13** [comment]
- https://www.sqlite.org/lang_mathfunc.html
- https://stackoverflow.com/questions/70451170/sqlite3-math-functions-python

Hmmmm

**Comment by damies13** [comment]
```
import sqlite3
db = sqlite3.connect("tmp.db")
c = self.datadb.cursor()
c.execute("PRAGMA compile_options;")
co.fetchall()
```

```
[
	('ATOMIC_INTRINSICS=1',), 
	('COMPILER=gcc-11.4.0',), 
	('DEFAULT_AUTOVACUUM',), 
	('DEFAULT_CACHE_SIZE=-2000',), 
	('DEFAULT_FILE_FORMAT=4',), 
	('DEFAULT_JOURNAL_SIZE_LIMIT=-1',), 
	('DEFAULT_MMAP_SIZE=0',), 
	('DEFAULT_PAGE_SIZE=4096',), 
	('DEFAULT_PCACHE_INITSZ=20',), 
	('DEFAULT_RECURSIVE_TRIGGERS',), 
	('DEFAULT_SECTOR_SIZE=4096',), 
	('DEFAULT_SYNCHRONOUS=2',), 
	('DEFAULT_WAL_AUTOCHECKPOINT=1000',), 
	('DEFAULT_WAL_SYNCHRONOUS=2',), 
	('DEFAULT_WORKER_THREADS=0',), 
	('ENABLE_COLUMN_METADATA',), 
	('ENABLE_DBSTAT_VTAB',), 
	('ENABLE_FTS3',), 
	('ENABLE_FTS3_PARENTHESIS',), 
	('ENABLE_FTS3_TOKENIZER',), 
	('ENABLE_FTS4',), 
	('ENABLE_FTS5',), 
	('ENABLE_JSON1',), 
	('ENABLE_LOAD_EXTENSION',), 
	('ENABLE_MATH_FUNCTIONS',), 
	('ENABLE_PREUPDATE_HOOK',), 
	('ENABLE_RTREE',), 
	('ENABLE_SESSION',), 
	('ENABLE_STMTVTAB',), 
	('ENABLE_UNLOCK_NOTIFY',), 
	('ENABLE_UPDATE_DELETE_LIMIT',), 
	('HAVE_ISNAN',), 
	('LIKE_DOESNT_MATCH_BLOBS',), 
	('MALLOC_SOFT_LIMIT=1024',), 
	('MAX_ATTACHED=10',), 
	('MAX_COLUMN=2000',), 
	('MAX_COMPOUND_SELECT=500',), 
	('MAX_DEFAULT_PAGE_SIZE=32768',), 
	('MAX_EXPR_DEPTH=1000',), 
	('MAX_FUNCTION_ARG=127',), 
	('MAX_LENGTH=1000000000',), 
	('MAX_LIKE_PATTERN_LENGTH=50000',), 
	('MAX_MMAP_SIZE=0x7fff0000',), 
	('MAX_PAGE_COUNT=1073741823',), 
	('MAX_PAGE_SIZE=65536',), 
	('MAX_SCHEMA_RETRY=25',), 
	('MAX_SQL_LENGTH=1000000000',), 
	('MAX_TRIGGER_DEPTH=1000',), 
	('MAX_VARIABLE_NUMBER=250000',), 
	('MAX_VDBE_OP=250000000',), 
	('MAX_WORKER_THREADS=8',), 
	('MUTEX_PTHREADS',), 
	('OMIT_LOOKASIDE',), 
	('SECURE_DELETE',), 
	('SOUNDEX',), 
	('SYSTEM_MALLOC',), 
	('TEMP_STORE=1',), 
	('THREADSAFE=1',), 
	('USE_URI',)
]
```


We need to check for `	('ENABLE_MATH_FUNCTIONS',), `

**Comment by damies13** [comment]
If it's not there we need to use [create_function](https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.create_function) to create the functions we need

**Comment by damies13** [comment]
Don't need the cursor, here's a simpler version
```
>>> import sqlite3
>>> 
>>> db = sqlite3.connect("tmp.db")
>>> co = db.execute("PRAGMA compile_options;")
>>> co.fetchall()
```

**Comment by damies13** [comment]
I've added manager tag as this could potentially impact manager as well

**Comment by damies13** [comment]
Fix for this issue merged into release branch 1.4.0
